tools/codegen/api/autograd.py

```diff
@@ -67,6 +67,14 @@ class ForwardDerivative:
     # Inputs for which the primal is required for this formula
     required_inputs_primal: Optional[Tuple[str, ...]]
 
+    # Flag to specify if this formula requires the original value of self
+    # This is only used by inplace operations
+    required_original_self_value: bool
+
+    # If this formula is specified by the user or if we are re-using the
```

Comment:
nit: "specified by the user" -> "specified in derivatives.yaml"? A user of PyTorch cannot specify a formula, right?